src/opnsense/mvc/app/controllers/OPNsense/Firewall/Api/FilterController.php

```diff
@@ -27,17 +27,131 @@
  */
 namespace OPNsense\Firewall\Api;
 
+use OPNsense\Core\Config;
+use OPNsense\Core\Backend;
+use OPNsense\Firewall\Category;
+use OPNsense\Firewall\Group;
+
+
 class FilterController extends FilterBaseController
 {
     protected static $categorysource = "rules.rule";
 
+    private function getFieldMap()
+    {
+        $result = ['category' => [], 'interface' => []];
+        foreach ((new Category())->categories->category->iterateItems() as $key => $category) {
+            $result['category'][$key] = (string)$category->name;
+        }
+        foreach ((Config::getInstance()->object())->interfaces->children() as $key => $ifdetail) {
+            $descr = htmlspecialchars(!empty($ifdetail->descr) ? $ifdetail->descr : strtoupper($ifname));
```

Comment:
This line crashes for me:

[09-Mar-2025 18:29:43 Etc/UTC] ErrorException: Undefined variable $ifname in /usr/local/opnsense/mvc/app/controllers/OPNsense/Firewall/Api/FilterController.php:47

Maybe like this?

```suggestion
            $descr = htmlspecialchars(!empty($ifdetail->descr) ? $ifdetail->descr : strtoupper($key));
```

Reply:
@Monviech got it, thanks, the `htmlspecialchars()` shouldn't be needed either. I'll push a fix in the next commit